### DISCUSSION TITLE
Tkinter app using Tensorflow...help needed.

### DISCUSSION THREAD

**Discussion by TNNSLAB** [pyinstaller/pyinstaller#8020] [Help] [answered]
Failing to build my app...help needed. Do I need a spec file ? 

My application folder contains this files :

vendor.py
transfers.py
requirements.txt
model_crop.py
movenet_lightning_fp16.tflite
model_crop_usa.py
icon2.png
grey_background.py
download.py
crop.py
app.py

**requirements.txt:**

beautifulsoup4==4.12.2
clipboard==0.0.4
numpy==1.22.3
opencv_python==4.5.1.48
Pillow==10.1.0
PyInstaller==6.1.0
requests==2.31.0
tensorflow==2.10.0

**My OS :**
Mojave 10.14.6
Python 3.9.6

my build command :  pyinstaller --onefile --windowed app.py

**Comment by rokm** [comment]
> Failing to build my app...help needed.

So what kind of error are you getting?

> icon2.png
> movenet_lightning_fp16.tflite

You will likely want to collect these files into the frozen application using `--add-data` or equivalent.

>  Mojave 10.14.6

That's unsupported version of macOS, and building on it might not work (depending on how buggy `codesign` is in that version).

> my build command : pyinstaller --onefile --windowed app.py

You should not be building onefile .app bundles.

**Reply by TNNSLAB** [reply]
wow, that was fast !

lots of errors...

[log.txt](https://github.com/pyinstaller/pyinstaller/files/13040257/log.txt)

what would you suggest to use an older OS system ? 

and how would my build command look like in the end ? 

pyinstaller --add-data --windowed app.py

**Reply by rokm** [reply, marked as answer]
> 554 INFO: PyInstaller: 4.8

That's an old and unsupported version of PyInstaller, too :) So if you think you're using 6.1.0 as per original report, you probably have python environments mixed up.

> ```
>Traceback (most recent call last):
>  File "<string>", line 3, in <module>
>  File "/usr/local/lib/python3.9/site-packages/pkg_resources/__init__.py", line 93, in <module>
>    __import__('pkg_resources.extern.packaging.requirements')
>  File "/usr/local/lib/python3.9/site-packages/pkg_resources/_vendor/packaging/requirements.py", line 8, in <module>
>    from ._parser import parse_requirement
>  File "/usr/local/lib/python3.9/site-packages/pkg_resources/_vendor/packaging/_parser.py", line 10, in <module>
>    from ._tokenizer import DEFAULT_RULES, Tokenizer
>  File "/usr/local/lib/python3.9/site-packages/pkg_resources/_vendor/packaging/_tokenizer.py", line 10, in <module>
>    class Token:
>  File "/usr/local/lib/python3.9/site-packages/dataclasses.py", line 958, in dataclass
>    return wrap(_cls)
>  File "/usr/local/lib/python3.9/site-packages/dataclasses.py", line 950, in wrap
>    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
>  File "/usr/local/lib/python3.9/site-packages/dataclasses.py", line 800, in _process_class
>    cls_fields = [_get_field(cls, name, type)
>  File "/usr/local/lib/python3.9/site-packages/dataclasses.py", line 800, in <listcomp>
>    cls_fields = [_get_field(cls, name, type)
>  File "/usr/local/lib/python3.9/site-packages/dataclasses.py", line 659, in _get_field
>    if (_is_classvar(a_type, typing)
>  File "/usr/local/lib/python3.9/site-packages/dataclasses.py", line 550, in _is_classvar
>    return type(a_type) is typing._ClassVar
>AttributeError: module 'typing' has no attribute '_ClassVar'
>``` 

These look like a problem with your python environment. `dataclasses` should be part of stdlib in python 3.9, so that `/usr/local/lib/python3.9/site-packages/dataclasses.py` looks like an external version that is incompatible with stdlib's `typing`. Assuming you have [this `dataclasses` package installed](https://pypi.org/project/dataclasses/), you should remove it (`pip uninstall dataclasses`).

> what would you suggest to use an older OS system ?

I have no suggestion, really, because `codesign` became integral to the build process due to Apple's signing requirements on modern systems (even for ad-hoc signing). We know that 10.15 is okay, and we know that 10.13 is not - 10.14 might or might not work. If you run into codesign errors and do not care about code signing (even ad-hoc), you can comment out the corresponding parts of the PyInstaller code. But from your current log, it looks like the `codesign` is working as expected.

> and how would my build command look like in the end ?
> 
> pyinstaller --add-data --windowed app.py

Take a look at documentation for [`--add-data`](https://pyinstaller.org/en/stable/usage.html#cmdoption-add-data) and [relevant](https://pyinstaller.org/en/stable/spec-files.html#adding-data-files) documentation [sections](https://pyinstaller.org/en/stable/runtime-information.html#placing-data-files-at-expected-locations-inside-the-bundle).

**Comment by TNNSLAB** [comment]
Ohhh ! Pretty messy ! 

- i removed the dataclasses. 
- i added my data into .spec file like : 
a = Analysis(['app.py'],
             pathex=['Users/adrian/Desktop/Imaging_App'],
             binaries=[],
             datas=['/Users/adrian/Desktop/Imaging_App/movenet_lightning_fp16.tflite']
             
             
             **Building the app like : pyinstaller app.spec**

  **and getting this error :** 
  
  335 INFO: Platform: macOS-10.14.6-x86_64-i386-64bit
341 INFO: Extending PYTHONPATH with paths
['/Users/adrian/Desktop/Imaging_App',
 '/Users/adrian/Desktop/Imaging_App/Users/adrian/Desktop/Imaging_App']
2006 INFO: Appending 'datas' from .spec
Traceback (most recent call last):
  File "/usr/local/bin/pyinstaller", line 8, in <module>
    sys.exit(_console_script_run())
  File "/usr/local/lib/python3.9/site-packages/PyInstaller/__main__.py", line 205, in _console_script_run
    run()
  File "/usr/local/lib/python3.9/site-packages/PyInstaller/__main__.py", line 189, in run
    run_build(pyi_config, spec_file, **vars(args))
  File "/usr/local/lib/python3.9/site-packages/PyInstaller/__main__.py", line 61, in run_build
    PyInstaller.building.build_main.main(pyi_config, spec_file, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 1042, in main
    build(specfile, distpath, workpath, clean_build)
  File "/usr/local/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 982, in build
    exec(code, spec_namespace)
  File "app.spec", line 7, in <module>
    a = Analysis(['app.py'],
  File "/usr/local/lib/python3.9/site-packages/PyInstaller/building/build_main.py", line 465, in __init__
    for dest_name, src_name in format_binaries_and_datas(datas, workingdir=spec_dir)]
  File "/usr/local/lib/python3.9/site-packages/PyInstaller/building/utils.py", line 420, in format_binaries_and_datas
    for src_root_path_or_glob, trg_root_dir in binaries_or_datas:
ValueError: too many values to unpack (expected 2)

any ideas ?

**Reply by rokm** [reply]
Please read the documentation again; the `datas` list expects tuples in form of `(SRC, DEST)`. So that
>  datas=['/Users/adrian/Desktop/Imaging_App/movenet_lightning_fp16.tflite']

should be

`datas=[('/Users/adrian/Desktop/Imaging_App/movenet_lightning_fp16.tflite', '.')]`

**Comment by TNNSLAB** [comment]
alright, that looks better but i'm getting this at the end : 

```
70377 INFO: Processing pre-safe import module hook win32com from '/usr/local/lib/python3.9/site-packages/_pyinstaller_hooks_contrib/hooks/pre_safe_import_module/hook-win32com.py'.

=============================================================
A RecursionError (maximum recursion depth exceeded) occurred.
For working around please follow these instructions
=============================================================

1. In your program's .spec file add this line near the top::

     import sys ; sys.setrecursionlimit(sys.getrecursionlimit() * 5)

2. Build your program by running PyInstaller with the .spec file as
   argument::

     pyinstaller myprog.spec

3. If this fails, you most probably hit an endless recursion in
   PyInstaller. Please try to track this down has far as possible,
   create a minimal example so we can reproduce and open an issue at
   https://github.com/pyinstaller/pyinstaller/issues following the
   instructions in the issue template. Many thanks.

Explanation: Python's stack-limit is a safety-belt against endless recursion,
eating up memory. PyInstaller imports modules recursively. If the structure
how modules are imported within your program is awkward, this leads to the
nesting being too deep and hitting Python's stack-limit.

With the default recursion limit (1000), the recursion error occurs at about
115 nested imported, with limit 2000 at about 240, with limit 5000 at about
660.
```

What could cause that ?

**Reply by rokm** [reply]
> What could cause that ?

Have you read the text? You need to raise the recursion limit in your spec file, as described...

**Reply by TNNSLAB** [reply]
Please don't get mad, I'm just trying to figure this one out...

So I fixed that recursionlimit by adding : 
  
  import sys
sys.setrecursionlimit(5000) inside my spec file 

the app was successfully build but when trying to open it...i'm getting this : 


File "app.py", line 15, in <module>
  File "PyInstaller/loader/pyimod02_importers.py", line 391, in exec_module
  File "PIL/Image.py", line 82, in <module>
ImportError: dlopen(/Users/adrian/Desktop/Imaging_App/dist/app/_internal/PIL/_imaging.cpython-39-darwin.so, 2): Library not loaded: @rpath/liblzma.5.dylib
  Referenced from: /Users/adrian/Desktop/Imaging_App/dist/app/_internal/PIL/.dylibs/libtiff.6.dylib
  Reason: Incompatible library version: libtiff.6.dylib requires version 10.0.0 or later, but liblzma.5.dylib provides version 8.0.0
[10422] Failed to execute script 'app' due to unhandled exception!

What should I do here ?

**Reply by rokm** [reply]
You have two packages that provide different versions of `liblzma.5.dylib`, and due to the way PyInstaller works, only one can be collected (or linked) into top-level application directory. I suppose you could manually try linking the other copy, and see if that works. Otherwise, you'll have to find versions of packages that bundle matching versions of dylibs (assuming such versions exist in the first place).

**Reply by TNNSLAB** [reply]
here's what i got : 

<img width="1194" alt="Screenshot 2023-10-19 at 13 35 05" src="https://github.com/pyinstaller/pyinstaller/assets/46636486/18555941-87b6-4e31-a1ce-2c8d43d3fe60">

How I can manually link it ? And where ?

**Reply by rokm** [reply]
There's should also be a `liblzma.5.dylib` symlink in the top-level `_internal` directory - where does it point?

**Reply by TNNSLAB** [reply]
yup...there's one there : 
<img width="886" alt="Screenshot 2023-10-19 at 13 43 22" src="https://github.com/pyinstaller/pyinstaller/assets/46636486/1b64a801-3b91-489a-be33-f0f34486f506">

**Reply by rokm** [reply]
And where does it point? (Since you'll likely have to remove this link and create new one, you should check in the terminal...)

**Reply by TNNSLAB** [reply]
when i drag that one into terminal i get : 
/Users/adrian/Desktop/Imaging_App/dist/app/_internal/liblzma.5.dylib

**Reply by rokm** [reply]
Mmm, fair enough. What if you open a terminal, and run:
```
cd /Users/adrian/Desktop/Imaging_App/dist/app/_internal
ls -l
```
Let's see the whole output.

**Reply by TNNSLAB** [reply]
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:49 AppKit
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:49 CoreFoundation
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:49 Foundation
drwxr-xr-x   7 adrian  staff      224 Oct 19 13:49 IPython
drwxr-xr-x   8 adrian  staff      256 Oct 19 13:49 PIL
drwxr-xr-x   7 adrian  staff      224 Oct 19 13:49 PyQt5
lrwxr-xr-x   1 adrian  staff       36 Oct 19 13:50 Python -> Python.framework/Versions/3.9/Python
drwxr-xr-x   5 adrian  staff      160 Oct 19 13:50 Python.framework
lrwxr-xr-x   1 adrian  staff       48 Oct 19 13:50 QtCore -> PyQt5/Qt5/lib/QtCore.framework/Versions/5/QtCore
lrwxr-xr-x   1 adrian  staff       48 Oct 19 13:50 QtDBus -> PyQt5/Qt5/lib/QtDBus.framework/Versions/5/QtDBus
lrwxr-xr-x   1 adrian  staff       46 Oct 19 13:50 QtGui -> PyQt5/Qt5/lib/QtGui.framework/Versions/5/QtGui
lrwxr-xr-x   1 adrian  staff       54 Oct 19 13:50 QtNetwork -> PyQt5/Qt5/lib/QtNetwork.framework/Versions/5/QtNetwork
lrwxr-xr-x   1 adrian  staff       64 Oct 19 13:50 QtPrintSupport -> PyQt5/Qt5/lib/QtPrintSupport.framework/Versions/5/QtPrintSupport
lrwxr-xr-x   1 adrian  staff       46 Oct 19 13:50 QtQml -> PyQt5/Qt5/lib/QtQml.framework/Versions/5/QtQml
lrwxr-xr-x   1 adrian  staff       58 Oct 19 13:50 QtQmlModels -> PyQt5/Qt5/lib/QtQmlModels.framework/Versions/5/QtQmlModels
lrwxr-xr-x   1 adrian  staff       50 Oct 19 13:50 QtQuick -> PyQt5/Qt5/lib/QtQuick.framework/Versions/5/QtQuick
lrwxr-xr-x   1 adrian  staff       46 Oct 19 13:50 QtSvg -> PyQt5/Qt5/lib/QtSvg.framework/Versions/5/QtSvg
lrwxr-xr-x   1 adrian  staff       60 Oct 19 13:50 QtWebSockets -> PyQt5/Qt5/lib/QtWebSockets.framework/Versions/5/QtWebSockets
lrwxr-xr-x   1 adrian  staff       54 Oct 19 13:50 QtWidgets -> PyQt5/Qt5/lib/QtWidgets.framework/Versions/5/QtWidgets
-rwxr-xr-x   1 adrian  staff  1011248 Oct 19 13:49 _brotli.cpython-39-darwin.so
-rwxr-xr-x   1 adrian  staff   223456 Oct 19 13:49 _cffi_backend.cpython-39-darwin.so
lrwxr-xr-x   1 adrian  staff       48 Oct 19 13:50 _pywrap_tensorflow_internal.so -> tensorflow/python/_pywrap_tensorflow_internal.so
-rwxr-xr-x   1 adrian  staff    86192 Oct 19 13:49 _tkinter.cpython-39-darwin.so
drwxr-xr-x   7 adrian  staff      224 Oct 19 13:50 attrs-23.1.0.dist-info
-rw-r--r--   1 adrian  staff   847390 Oct 19 13:49 base_library.zip
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:50 boto3
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:50 botocore
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 cchardet
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:49 certifi
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:49 charset_normalizer
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 contourpy
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 cryptography
drwxr-xr-x  11 adrian  staff      352 Oct 19 13:49 cryptography-40.0.2.dist-info
drwxr-xr-x   7 adrian  staff      224 Oct 19 13:50 cv2
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 google
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 greenlet
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 grpc
drwxr-xr-x  27 adrian  staff      864 Oct 19 13:49 h5py
drwxr-xr-x   8 adrian  staff      256 Oct 19 13:49 importlib_metadata-6.3.0.dist-info
drwxr-xr-x   7 adrian  staff      224 Oct 19 13:49 jaxlib
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 jedi
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 jsonschema
drwxr-xr-x   8 adrian  staff      256 Oct 19 13:49 jsonschema-4.19.1.dist-info
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 jsonschema_specifications
-rwxr-xr-x   1 adrian  staff   238096 Oct 19 13:49 kiwisolver.cpython-39-darwin.so
drwxr-xr-x  60 adrian  staff     1920 Oct 19 13:49 lib-dynload
lrwxr-xr-x   1 adrian  staff       30 Oct 19 13:50 libXau.6.0.0.dylib -> PIL/.dylibs/libXau.6.0.0.dylib
lrwxr-xr-x   1 adrian  staff       27 Oct 19 13:50 libarrow.1200.dylib -> pyarrow/libarrow.1200.dylib
lrwxr-xr-x   1 adrian  staff       33 Oct 19 13:50 libarrow_acero.1200.dylib -> pyarrow/libarrow_acero.1200.dylib
lrwxr-xr-x   1 adrian  staff       35 Oct 19 13:50 libarrow_dataset.1200.dylib -> pyarrow/libarrow_dataset.1200.dylib
lrwxr-xr-x   1 adrian  staff       34 Oct 19 13:50 libarrow_flight.1200.dylib -> pyarrow/libarrow_flight.1200.dylib
lrwxr-xr-x   1 adrian  staff       29 Oct 19 13:50 libarrow_python.dylib -> pyarrow/libarrow_python.dylib
lrwxr-xr-x   1 adrian  staff       38 Oct 19 13:50 libavcodec.58.54.100.dylib -> cv2/.dylibs/libavcodec.58.54.100.dylib
lrwxr-xr-x   1 adrian  staff       39 Oct 19 13:50 libavformat.58.29.100.dylib -> cv2/.dylibs/libavformat.58.29.100.dylib
lrwxr-xr-x   1 adrian  staff       37 Oct 19 13:50 libavresample.4.0.0.dylib -> cv2/.dylibs/libavresample.4.0.0.dylib
lrwxr-xr-x   1 adrian  staff       37 Oct 19 13:50 libavutil.56.31.100.dylib -> cv2/.dylibs/libavutil.56.31.100.dylib
lrwxr-xr-x   1 adrian  staff       29 Oct 19 13:50 libbluray.2.dylib -> cv2/.dylibs/libbluray.2.dylib
lrwxr-xr-x   1 adrian  staff       39 Oct 19 13:50 libbrotlicommon.1.1.0.dylib -> PIL/.dylibs/libbrotlicommon.1.1.0.dylib
lrwxr-xr-x   1 adrian  staff       36 Oct 19 13:50 libbrotlidec.1.1.0.dylib -> PIL/.dylibs/libbrotlidec.1.1.0.dylib
lrwxr-xr-x   1 adrian  staff       36 Oct 19 13:50 libc++.1.0.dylib -> torchvision/.dylibs/libc++.1.0.dylib
lrwxr-xr-x   1 adrian  staff       22 Oct 19 13:50 libc10.dylib -> torch/lib/libc10.dylib
lrwxr-xr-x   1 adrian  staff       31 Oct 19 13:50 libcrypto.1.1.dylib -> cv2/.dylibs/libcrypto.1.1.dylib
lrwxr-xr-x   1 adrian  staff       26 Oct 19 13:50 libffi.6.dylib -> cv2/.dylibs/libffi.6.dylib
lrwxr-xr-x   1 adrian  staff       33 Oct 19 13:50 libfontconfig.1.dylib -> cv2/.dylibs/libfontconfig.1.dylib
lrwxr-xr-x   1 adrian  staff       31 Oct 19 13:50 libfreetype.6.dylib -> PIL/.dylibs/libfreetype.6.dylib
lrwxr-xr-x   1 adrian  staff       32 Oct 19 13:50 libgcc_s.1.1.dylib -> scipy/.dylibs/libgcc_s.1.1.dylib
lrwxr-xr-x   1 adrian  staff       30 Oct 19 13:50 libgcc_s.1.dylib -> numpy/.dylibs/libgcc_s.1.dylib
lrwxr-xr-x   1 adrian  staff       33 Oct 19 13:50 libgfortran.3.dylib -> numpy/.dylibs/libgfortran.3.dylib
lrwxr-xr-x   1 adrian  staff       33 Oct 19 13:50 libgfortran.5.dylib -> scipy/.dylibs/libgfortran.5.dylib
lrwxr-xr-x   1 adrian  staff       27 Oct 19 13:50 libgmp.10.dylib -> cv2/.dylibs/libgmp.10.dylib
lrwxr-xr-x   1 adrian  staff       30 Oct 19 13:50 libgnutls.30.dylib -> cv2/.dylibs/libgnutls.30.dylib
lrwxr-xr-x   1 adrian  staff       31 Oct 19 13:50 libharfbuzz.0.dylib -> PIL/.dylibs/libharfbuzz.0.dylib
lrwxr-xr-x   1 adrian  staff       30 Oct 19 13:50 libhdf5.200.dylib -> h5py/.dylibs/libhdf5.200.dylib
lrwxr-xr-x   1 adrian  staff       33 Oct 19 13:50 libhdf5_hl.200.dylib -> h5py/.dylibs/libhdf5_hl.200.dylib
lrwxr-xr-x   1 adrian  staff       32 Oct 19 13:50 libhogweed.4.5.dylib -> cv2/.dylibs/libhogweed.4.5.dylib
lrwxr-xr-x   1 adrian  staff       27 Oct 19 13:50 libidn2.0.dylib -> cv2/.dylibs/libidn2.0.dylib
lrwxr-xr-x   1 adrian  staff       27 Oct 19 13:50 libintl.8.dylib -> cv2/.dylibs/libintl.8.dylib
lrwxr-xr-x   1 adrian  staff       28 Oct 19 13:50 libiomp5.dylib -> torch/.dylibs/libiomp5.dylib
lrwxr-xr-x   1 adrian  staff       48 Oct 19 13:50 libjaxlib_mlir_capi.dylib -> jaxlib/mlir/_mlir_libs/libjaxlib_mlir_capi.dylib
lrwxr-xr-x   1 adrian  staff       28 Oct 19 13:50 liblcms2.2.dylib -> PIL/.dylibs/liblcms2.2.dylib
-rwxr-xr-x   1 adrian  staff   308624 Oct 19 13:49 liblzma.5.dylib
lrwxr-xr-x   1 adrian  staff       30 Oct 19 13:50 libmp3lame.0.dylib -> cv2/.dylibs/libmp3lame.0.dylib
-rwxr-xr-x   1 adrian  staff   159984 Oct 19 13:49 libmpdec.3.dylib
lrwxr-xr-x   1 adrian  staff       31 Oct 19 13:50 libnettle.6.5.dylib -> cv2/.dylibs/libnettle.6.5.dylib
lrwxr-xr-x   1 adrian  staff       26 Oct 19 13:50 libogg.0.dylib -> cv2/.dylibs/libogg.0.dylib
lrwxr-xr-x   1 adrian  staff       33 Oct 19 13:50 libopenblas.0.dylib -> scipy/.dylibs/libopenblas.0.dylib
lrwxr-xr-x   1 adrian  staff       36 Oct 19 13:50 libopenblas64_.0.dylib -> numpy/.dylibs/libopenblas64_.0.dylib
lrwxr-xr-x   1 adrian  staff       37 Oct 19 13:50 libopencore-amrnb.0.dylib -> cv2/.dylibs/libopencore-amrnb.0.dylib
lrwxr-xr-x   1 adrian  staff       37 Oct 19 13:50 libopencore-amrwb.0.dylib -> cv2/.dylibs/libopencore-amrwb.0.dylib
lrwxr-xr-x   1 adrian  staff       34 Oct 19 13:50 libopenjp2.2.3.1.dylib -> cv2/.dylibs/libopenjp2.2.3.1.dylib
lrwxr-xr-x   1 adrian  staff       34 Oct 19 13:50 libopenjp2.2.5.0.dylib -> PIL/.dylibs/libopenjp2.2.5.0.dylib
lrwxr-xr-x   1 adrian  staff       27 Oct 19 13:50 libopus.0.dylib -> cv2/.dylibs/libopus.0.dylib
lrwxr-xr-x   1 adrian  staff       30 Oct 19 13:50 libp11-kit.0.dylib -> cv2/.dylibs/libp11-kit.0.dylib
lrwxr-xr-x   1 adrian  staff       29 Oct 19 13:50 libparquet.1200.dylib -> pyarrow/libparquet.1200.dylib
lrwxr-xr-x   1 adrian  staff       29 Oct 19 13:50 libpng16.16.dylib -> PIL/.dylibs/libpng16.16.dylib
lrwxr-xr-x   1 adrian  staff       33 Oct 19 13:50 libquadmath.0.dylib -> numpy/.dylibs/libquadmath.0.dylib
-rwxr-xr-x   1 adrian  staff   265648 Oct 19 13:49 libreadline.8.dylib
lrwxr-xr-x   1 adrian  staff       27 Oct 19 13:50 librtmp.1.dylib -> cv2/.dylibs/librtmp.1.dylib
lrwxr-xr-x   1 adrian  staff       31 Oct 19 13:50 libsharpyuv.0.dylib -> PIL/.dylibs/libsharpyuv.0.dylib
lrwxr-xr-x   1 adrian  staff       22 Oct 19 13:50 libshm.dylib -> torch/lib/libshm.dylib
lrwxr-xr-x   1 adrian  staff       33 Oct 19 13:50 libsnappy.1.1.7.dylib -> cv2/.dylibs/libsnappy.1.1.7.dylib
lrwxr-xr-x   1 adrian  staff       30 Oct 19 13:50 libsodium.23.dylib -> zmq/.dylibs/libsodium.23.dylib
lrwxr-xr-x   1 adrian  staff       31 Oct 19 13:50 libsoxr.0.1.2.dylib -> cv2/.dylibs/libsoxr.0.1.2.dylib
lrwxr-xr-x   1 adrian  staff       28 Oct 19 13:50 libspeex.1.dylib -> cv2/.dylibs/libspeex.1.dylib
-rwxr-xr-x   1 adrian  staff  1087920 Oct 19 13:49 libsqlite3.0.dylib
lrwxr-xr-x   1 adrian  staff       28 Oct 19 13:50 libssl.1.1.dylib -> cv2/.dylibs/libssl.1.1.dylib
lrwxr-xr-x   1 adrian  staff       39 Oct 19 13:50 libswresample.3.5.100.dylib -> cv2/.dylibs/libswresample.3.5.100.dylib
lrwxr-xr-x   1 adrian  staff       36 Oct 19 13:50 libswscale.5.5.100.dylib -> cv2/.dylibs/libswscale.5.5.100.dylib
lrwxr-xr-x   1 adrian  staff       28 Oct 19 13:50 libtasn1.6.dylib -> cv2/.dylibs/libtasn1.6.dylib
-rwxr-xr-x   1 adrian  staff  1427776 Oct 19 13:49 libtcl8.6.dylib
lrwxr-xr-x   1 adrian  staff       42 Oct 19 13:50 libtensorflow_framework.2.dylib -> tensorflow/libtensorflow_framework.2.dylib
lrwxr-xr-x   1 adrian  staff       32 Oct 19 13:50 libtheoradec.1.dylib -> cv2/.dylibs/libtheoradec.1.dylib
lrwxr-xr-x   1 adrian  staff       32 Oct 19 13:50 libtheoraenc.1.dylib -> cv2/.dylibs/libtheoraenc.1.dylib
lrwxr-xr-x   1 adrian  staff       27 Oct 19 13:50 libtiff.6.dylib -> PIL/.dylibs/libtiff.6.dylib
-rwxr-xr-x   1 adrian  staff  1448864 Oct 19 13:49 libtk8.6.dylib
lrwxr-xr-x   1 adrian  staff       24 Oct 19 13:50 libtorch.dylib -> torch/lib/libtorch.dylib
lrwxr-xr-x   1 adrian  staff       28 Oct 19 13:50 libtorch_cpu.dylib -> torch/lib/libtorch_cpu.dylib
lrwxr-xr-x   1 adrian  staff       31 Oct 19 13:50 libtorch_python.dylib -> torch/lib/libtorch_python.dylib
lrwxr-xr-x   1 adrian  staff       32 Oct 19 13:50 libunistring.2.dylib -> cv2/.dylibs/libunistring.2.dylib
lrwxr-xr-x   1 adrian  staff       29 Oct 19 13:50 libvorbis.0.dylib -> cv2/.dylibs/libvorbis.0.dylib
lrwxr-xr-x   1 adrian  staff       32 Oct 19 13:50 libvorbisenc.2.dylib -> cv2/.dylibs/libvorbisenc.2.dylib
lrwxr-xr-x   1 adrian  staff       27 Oct 19 13:50 libwebp.7.dylib -> PIL/.dylibs/libwebp.7.dylib
lrwxr-xr-x   1 adrian  staff       32 Oct 19 13:50 libwebpdemux.2.dylib -> PIL/.dylibs/libwebpdemux.2.dylib
lrwxr-xr-x   1 adrian  staff       30 Oct 19 13:50 libwebpmux.3.dylib -> PIL/.dylibs/libwebpmux.3.dylib
lrwxr-xr-x   1 adrian  staff       30 Oct 19 13:50 libxcb.1.1.0.dylib -> PIL/.dylibs/libxcb.1.1.0.dylib
lrwxr-xr-x   1 adrian  staff       30 Oct 19 13:50 libz.1.2.12.dylib -> h5py/.dylibs/libz.1.2.12.dylib
lrwxr-xr-x   1 adrian  staff       26 Oct 19 13:50 libz.1.3.dylib -> PIL/.dylibs/libz.1.3.dylib
lrwxr-xr-x   1 adrian  staff       26 Oct 19 13:50 libzmq.5.dylib -> zmq/.dylibs/libzmq.5.dylib
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 llvmlite
drwxr-xr-x   9 adrian  staff      288 Oct 19 13:49 lxml
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 markupsafe
drwxr-xr-x  10 adrian  staff      320 Oct 19 13:49 matplotlib
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 ml_dtypes
-rwxr-xr-x   1 adrian  staff  4758512 Oct 19 13:49 movenet_lightning_fp16.tflite
drwxr-xr-x   5 adrian  staff      160 Oct 19 13:49 nbformat
drwxr-xr-x   8 adrian  staff      256 Oct 19 13:50 nbformat-5.9.2.dist-info
drwxr-xr-x  11 adrian  staff      352 Oct 19 13:49 numba
drwxr-xr-x   7 adrian  staff      224 Oct 19 13:49 numpy
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 objc
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:49 pandas
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:49 parso
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:49 psutil
drwxr-xr-x  77 adrian  staff     2464 Oct 19 13:50 pyarrow
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 pycocotools
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:50 pytz
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 regex
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 rpds
drwxr-xr-x  17 adrian  staff      544 Oct 19 13:49 scipy
drwxr-xr-x   5 adrian  staff      160 Oct 19 13:49 sqlalchemy
drwxr-xr-x  41 adrian  staff     1312 Oct 19 13:50 tcl
drwxr-xr-x   5 adrian  staff      160 Oct 19 13:50 tcl8
drwxr-xr-x   6 adrian  staff      192 Oct 19 13:49 tensorflow
drwxr-xr-x  39 adrian  staff     1248 Oct 19 13:50 tk
drwxr-xr-x  69 adrian  staff     2208 Oct 19 13:50 torch
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 torchaudio
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:49 torchvision
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 tornado
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 uvloop
drwxr-xr-x   8 adrian  staff      256 Oct 19 13:50 wheel-0.40.0.dist-info
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 wrapt
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 yaml
drwxr-xr-x   4 adrian  staff      128 Oct 19 13:49 zmq
drwxr-xr-x   3 adrian  staff       96 Oct 19 13:49 zope

**Reply by TNNSLAB** [reply]
looks like it's working now, the app starts and performs well...but the whole app is 2.3Gb ! even though i'm running inside venv 

any ideas ? is it the Tensorflow package that is so big ?

**Reply by rokm** [reply]
> looks like it's working now, the app starts and performs well...but the whole app is 2.3Gb ! even though i'm running inside venv
> 
> any ideas ? is it the Tensorflow package that is so big ?

Yeah, you have several large packages there - `tensorflow`, `PyQt5`, `scipy`, `numpy`...

**Comment by TNNSLAB** [comment]
I don't really use PyQt5...why is it in there ? or other apps are using it ? 
Thank you very much for helping me with this !

**Reply by rokm** [reply]
Because it is imported somewhere (and is available in the environment) - even if you do not use that codepath. If you are 100% that you don't need it, you can add `--exclude PyQt5` (or to `excludes` list that is passed to `Analysis` in the .spec file).

**Reply by TNNSLAB** [reply]
@rokm  you da best !
